stern/stern.go

```diff
@@ -185,50 +161,41 @@ func Run(ctx context.Context, config *Config) error {
 						return
 					}
 					added <- v
-				case v, ok := <-r:
-					if !ok {
-						errCh <- fmt.Errorf("lost watch connection")
-						return
-					}
-					removed <- v
 				case <-ctx.Done():
 					return
 				}
 			}
 		}()
 	}
 
-	go func() {
-		for p := range added {
-			targetID := p.GetID()
-
-			if tail, ok := getTail(targetID); ok {
-				if tail.isActive() {
-					continue
-				} else {
-					tail.Close()
-					clearTail(targetID)
-				}
+	addTarget := func(ctx context.Context, target *Target) {
+		for {
+			start := time.Now()
+			tail := newTail(target)
+			err := tail.Start(ctx)
+			tail.Close()
+			if err == nil {
+				return
 			}
-
-			tail := newTail(p)
-			setTail(targetID, tail)
-
-			go func(tail *Tail) {
-				if err := tail.Start(ctx); err != nil {
-					fmt.Fprintf(config.ErrOut, "unexpected error: %v\n", err)
+			if !filter.isActive(target) {
+				fmt.Fprintf(config.ErrOut, "failed to tail: %v\n", err)
+				return
+			}
+			if interval := time.Until(start.Add(1 * time.Minute)); interval > 0 {
+				fmt.Fprintf(config.ErrOut, "failed to tail: %v, will retry in %s\n", err, interval)
+				select {
+				case <-ctx.Done():
+				case <-time.After(interval):
 				}
-			}(tail)
+				return
```

Comment:
Since `addTarget` function exits here, it does not look like it is retrying.

Reply:
That's true. I will fix it after https://github.com/stern/stern/pull/221#discussion_r1096703044. Thanks!

Reply:
I've fixed it in https://github.com/stern/stern/commit/01dedcad56fbb135043e7ae8c71fba98ca38d1d8.